### PR TITLE
Move instantiating a Process inside the process manager

### PR DESCRIPTION
**Pull request by richardmiller-zz**
May be a controversial one :) 

Everything that wants to run a command using a process runner has to instantiate a process object to pass to the process manager. This moves that inside the process runners and lets the calling code just pass the command as a string.

### CODE REVIEW

File: src/Cli/IO/InteractiveProcessRunner.php [hunk 1]
```diff
@@ -55,7 +57,7 @@ private function getRunningProcessCallback($highlightErrors = true)
             foreach ($lines as $line) {
                 $line = trim($line);
                 if (!empty($line)) {
-                    $this->userInteraction->write($prefix.' '.$line);
+                    $this->userInteraction->write($prefix . ' ' . $line);
```
Comment (sroze) [inline src/Cli/IO/InteractiveProcessRunner.php]: You might have a strange PHPStorm configuration that breaks PSRs ;-)

File: src/Cli/UpCommand.php [hunk 2]
```diff
@@ -42,8 +41,7 @@ protected function configure()
     {
         $this
             ->setName('up')
-            ->setDescription('Start the project')
-        ;
```
Comment (sroze) [inline src/Cli/UpCommand.php:46]: Why removing that ? I'd say everything on the same line or everything on different lines.

**Comment by sroze** [comment]
Except my few comments, I definitely agree ! This remove the direct dependency to Symfony's Process component and will simplify the mocking process of the process runner :+1:

**Comment by richardmiller-zz** [comment]
I've updated the auto-formatted things.

File: src/Cli/IO/InteractiveProcessRunner.php (hunk 1 above)
Comment (richardmiller-zz) [inline src/Cli/IO/InteractiveProcessRunner.php]: I don't think the PSRs dictate this either way so your preference wins :)

File: src/IO/ProcessRunner.php [hunk 3]
```diff
@@ -7,10 +7,10 @@
 interface ProcessRunner
 {
     /**
-     * @param Process $process
-     * @param bool    $mustSucceed
+     * @param $command
```
Comment (sroze) [inline src/IO/ProcessRunner.php]: `string` ?

File: src/Cli/UpCommand.php (hunk 2 above)
Comment (richardmiller-zz) [inline src/Cli/UpCommand.php:46]: I think PSR-2 (or at least phpcs with that standard) doesn't like this. I prefer it the way it was though so undone my bad auto-formatting.

File: src/Cli/IO/InteractiveProcessRunner.php (hunk 1 above)
Comment (sroze) [inline src/Cli/IO/InteractiveProcessRunner.php]: \o/ maybe [you're right](https://github.com/symfony/symfony/issues/3135)

**Comment by sroze** [comment]
:+1: 
(we lost 0.08 on Scrutinizer btw..)

**Comment by richardmiller-zz** [comment]
Scrutinizer might because of the issues I've just opened #26 for.

**Comment by sroze** [comment]
Right, let's fix this after, it's a false positive. Thanks again :+1:

**Pull request merged**